pyvista/plotting/themes.py

```diff
@@ -230,6 +252,71 @@ def _all__slots__(cls):
         mro = cls.mro()
         return tuple(chain.from_iterable(c.__slots__ for c in mro if c is not object))  # type: ignore[attr-defined]
 
+    @classmethod
+    def register_default(cls, name, dict_, doc=None):
+        """Register a default theme from a dictionary.
+
+        Parameters
+        ----------
+        name : str
+            Name of theme for attribute.
+        dict_ : dict
+            Theme dictionary
+        doc : str, optional
+            Docstring to add to attribute.
+
+        Examples
+        --------
+        Register a default theme for downstream use.
+
+        >>> from pyvista.plotting.themes import Theme
+        >>> Theme.register_default("my_theme", {"show_edges": True})
+        >>> my_theme = Theme.my_theme()
+        >>> my_theme.show_edges
+        True
+
+        """
+
+        def return_theme(cls):
+            return cls.from_dict(dict_)
+
+        setattr(cls, name, classmethod(return_theme))
+        attr = getattr(cls, name)
+        attr.__func__.__name__ = name
+        if doc is not None:
+            attr.__func__.__doc__ = doc
+
+        cls._defaults[name] = name
+
+    @classmethod
+    def from_default(cls, name):
+        """Return a default theme.
+
+        Examples
+        --------
+        >>> from pyvista.plotting.themes import Theme
+        >>> theme = Theme.from_default("vtk")
+
+        See Also
+        --------
+        Theme.defaults
+
+        """
+        theme = getattr(cls, cls._defaults[name])
+        return theme()
+
+    @classmethod
+    def defaults(self):
+        """Return list of default themes.
```

Comment:
Maybe this can be used instead? Can be probably me moved into core utilities somewhere for re-use
https://github.com/pyvista/pyvista/blob/ed4178eefeb1553ffe2d902150285e648ae1c4c4/doc/source/make_tables.py#L85-L94

Reply:
Let's do this if possible!